Tests/DiffableDataSourceTests/DiffableTests.swift

```diff
@@ -2,10 +2,104 @@ import XCTest
+
+    lazy var content = [topSection, middleSection, bottomSection]
+
+    // MARK: - Overrides
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        dataSource = DiffableTableViewDataSource<Section, Row>(
+            tableView: UITableView(),
+            cellProvider: { _, _, _ in
+                nil
+            },
+            canEditProvider: { [unowned self] indexPath in
+                self.dataSource.section(at: indexPath.section) == .middle
+            }
+        )
+    }
+
+    func testReplaceEmpty() {
+        XCTAssertNoThrow(try dataSource.replace(with: []))
+        XCTAssertTrue(dataSource.snapshot().sectionIdentifiers.isEmpty)
+        XCTAssertTrue(dataSource.snapshot().itemIdentifiers.isEmpty)
+    }
+
+    func testReplaceWithNonUniqueSectionIdentifiers() {
+        XCTAssertThrowsError(try dataSource.replace(with: [topSection, topSection]))
     }
 
-    static var allTests = [
-        ("testExample", testExample),
-    ]
+    func testReplaceWithNonUniqueRowIdentifiers() {
+        let wrongBottomSection = DiffableSection<Section, Row>(identifier: .bottom, rows: [.headerElement])
+
+        XCTAssertThrowsError(
+            try dataSource.replace(with: [topSection, wrongBottomSection], animatingDifferences: false)
+        )
+    }
+
+    func testReplace() {
+        let content = [topSection, middleSection, bottomSection]
+
+        XCTAssertNoThrow(try dataSource.replace(with: content, animatingDifferences: false))
+
+        XCTAssert(dataSource.snapshot().sectionIdentifiers == content.map { $0.identifier })
+        XCTAssert(dataSource.snapshot().itemIdentifiers == content.flatMap { $0.rows })
+    }
+
+    func testCanEdit() {
+        XCTAssertNoThrow(try dataSource.replace(with: content, animatingDifferences: false))
+
+        XCTAssertFalse(dataSource.tableView(tableView, canEditRowAt: IndexPath(row: 0, section: 0)))
+        XCTAssertTrue(dataSource.tableView(tableView, canEditRowAt: IndexPath(row: 0, section: 1)))
+        XCTAssertTrue(dataSource.tableView(tableView, canEditRowAt: IndexPath(row: 1, section: 1)))
+        XCTAssertFalse(dataSource.tableView(tableView, canEditRowAt: IndexPath(row: 0, section: 2)))
+    }
+
+    func testSectionAt() {
+        XCTAssertNoThrow(try dataSource.replace(with: content, animatingDifferences: false))
+
+        for index in 0..<content.count {
+            XCTAssertTrue(dataSource.section(at: index) == content[index].identifier)
+        }
+    }
+
+    func testRowAt() {
+        let content = [topSection, middleSection, bottomSection]
+
+        XCTAssertNoThrow(try dataSource.replace(with: content, animatingDifferences: false))
+
+        for section in 0..<content.count {
+            for row in 0..<content[section].rows.count {
+                XCTAssertTrue(dataSource.row(at: IndexPath(row: row, section: section)) == content[section].rows[row])
```

Comment:
It's better to use XCTAssertEqual when you can as it will give more information if that assertion fails. 
Also when asserting with multiple values it's a good practice to provide more context in the error message as to which assertion failed. This can be very useful when trying to determine what failed during CI runs.
```suggestion
                let indexPath = IndexPath(row: row, section: section)
                XCTAssertEqual(
                    dataSource.row(at: indexPath),
                    content[section].rows[row],
                    "Row wrong at indexPath:(\(indexPath))"
                )
```